sway/input/keyboard.c

```diff
@@ -334,9 +336,52 @@ void sway_keyboard_disarm_key_repeat(struct sway_keyboard *keyboard) {
 	}
 }
 
+struct key_info {
+	uint32_t keycode;
+	uint32_t code_modifiers;
+
+	const xkb_keysym_t *raw_keysyms;
+	uint32_t raw_modifiers;
+	size_t raw_keysyms_len;
+
+	const xkb_keysym_t *translated_keysyms;
+	uint32_t translated_modifiers;
+	size_t translated_keysyms_len;
+};
+
+static void update_keyboard_state(struct sway_keyboard *keyboard,
+		uint32_t raw_keycode, enum wlr_key_state keystate,
+		struct key_info *keyinfo) {
+	keyinfo->keycode = raw_keycode + 8;
```

Comment:
I know this is just pre-existing code that was moved around, but what does this 8 represent?

Reply:
xkb keycodes are 8 higher than event codes (https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h). I'm blanking out on the reasoning.